### PR TITLE
Change way to display line from svg to gradient. Stored Force Height …

### PR DESCRIPTION
**Pull request by borracciaBlu**
- Change the way to display line from svg to gradient. Now we can use fractions of px.
- Stored Force Height Flag.

### CODE REVIEW

File: chrome_extension/baseliner.js [hunk 1]
```diff
@@ -52,29 +50,47 @@ Baseliner = {
...
+			g: parseInt(result[2], 16),
+			b: parseInt(result[3], 16)
+		} : null;
+	},
 
 	/**
 	 * Add CSS rules into Baseliner's styleSheet
 	 * @param background
 	 * @param top
 	 */
```
Comment (jpedroribeiro) [inline chrome_extension/baseliner.js:85]: could you update the @params to cater for the new parameters please

File: chrome_extension/baseliner.js [hunk 2]
```diff
@@ -164,11 +163,12 @@ Baseliner = {
 	 * @param top
 	 * @param opacity
 	 */
```
Comment (jpedroribeiro) [inline chrome_extension/baseliner.js:170]: same here

File: chrome_extension/main.js [hunk 3]
```diff
@@ -36,8 +37,8 @@ var updatesBaseliner = function(event){
     if (event.keyIdentifier === 'Down' && event.target === $extensionBaseline) $extensionBaseline.value--;
     if (event.key === "ArrowUp" && event.target === $extensionBaseline) $extensionBaseline.value++;
     if (event.key === "ArrowDown" && event.target === $extensionBaseline && $extensionBaseline.value > 2) $extensionBaseline.value--;
-    if (event.keyIdentifier === 'Up' && event.target === $extensionOpacity) $extensionOpacity.value++;
-    if (event.keyIdentifier === 'Down' && event.target === $extensionOpacity) $extensionOpacity.value--;
+    if (event.keyIdentifier === 'Up' && event.target === $extensionOpacity && $extensionOpacity.value < 100) $extensionOpacity.value++;
+    if (event.keyIdentifier === 'Down' && event.target === $extensionOpacity && $extensionOpacity.value > 0) $extensionOpacity.value--;
     if (event.key === "ArrowUp" && event.target === $extensionOpacity && $extensionOpacity.value < 100) $extensionOpacity.value++;
```
Comment (jpedroribeiro) [inline chrome_extension/main.js:42]: ah, well spotted!

File: chrome_extension/baseliner.js (hunk 1 above)
Comment (borracciaBlu) [reply chrome_extension/baseliner.js:85]: it should be fine now

File: chrome_extension/main.js (hunk 3 above)
Comment (borracciaBlu) [reply chrome_extension/main.js:42]: thanks

File: chrome_extension/styles.css [hunk 4]
```diff
@@ -607,7 +607,7 @@ header {
         font-size: 24px;
         color: #10174C;
         border: none;
-        width: 40px;
+        width: 60px;
         text-align: right;
```
Comment (jpedroribeiro) [inline chrome_extension/styles.css:612]: All these CSS updates are pushing the colour input to below the visible area.
Adding `white-space: nowrap;` to the .fieldInput will make sure it stays where it should.

File: chrome_extension/styles.css [hunk 5]
```diff
@@ -704,6 +704,13 @@ header {
     left: 30px!important;
 }
 
+.cp-opacity, .cp-memo {
```
Comment (jpedroribeiro) [inline chrome_extension/styles.css:716]: would you mind adding a comment stating that these styles are to overwrite the jscolor plugin classes and hide opacity menu from the popup.

**Review by jpedroribeiro — CHANGES_REQUESTED** [review]
I like the idea of using gradients instead. Much more performant and customisable I believe.
I've added some comments regarding CSS which seem to be hiding the colour input.

**Comment by borracciaBlu** [comment]
It should be fine now. Let me know. Thanks for the feedback.

**Review by jpedroribeiro — APPROVED** [review]

**Pull request merged**